src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -551,83 +581,64 @@ export async function install(
       }
-    }
-    let executable = path.join(destinationPath, platformInfo.executable)
 
-    const zipFile = path.join(directory, fileName)
-    logInfo(`Extracting ${zipFile} to ${destinationPath}`, LogPrefix.HyperPlay)
+      logInfo(`Installing ${title} to ${dirpath}...`, LogPrefix.HyperPlay)
+      const zipPathInfo = getZipFileName(appName, platformInfo)
+      directory = zipPathInfo.directory
+      fileName = zipPathInfo.filename
+
+      // download the zip file
+      if (!existsSync(destinationPath)) {
+        mkdirSync(destinationPath, { recursive: true })
+      }
 
-    try {
-      window.webContents.send('gameStatusUpdate', {
+      // Reset the download progress
+      window.webContents.send(`progressUpdate-${appName}`, {
         appName,
         runner: 'hyperplay',
         folder: destinationPath,
-        status: 'extracting'
+        status: 'done',
+        progress: {
+          folder: destinationPath,
+          percent: 0,
+          diskSpeed: 0,
+          downSpeed: 0,
+          bytes: 0,
+          eta: null
+        }
       })
 
-      // disables electron's fs wrapper called when extracting .asar files
-      // which is necessary to extract electron app/game zip files
-      process.noAsar = true
-      if (isWindows) {
-        await extractZip(zipFile, destinationPath)
-        await installDistributables(destinationPath)
-      } else {
-        await extractZip(zipFile, destinationPath)
-      }
-      process.noAsar = false
-
-      if (isMac && executable.endsWith('.app')) {
-        const macAppExecutable = readdirSync(
-          join(executable, 'Contents', 'MacOS')
-        )[0]
-        executable = join(executable, 'Contents', 'MacOS', macAppExecutable)
-      }
-
-      const installedInfo: InstalledInfo = {
+      await downloadGame(
         appName,
-        install_path: destinationPath,
-        executable: executable,
-        install_size: platformInfo.installSize ?? '0',
-        is_dlc: false,
-        version: installVersion,
-        platform: appPlatform,
-        channelName
-      }
-
-      updateInstalledInfo(appName, installedInfo)
-
-      notify({
-        title,
-        body: `Installed`
-      })
-
-      cleanUpDownload(appName, directory)
+        directory,
+        fileName,
+        platformInfo,
+        destinationPath
+      )
+    }
 
-      sendFrontendMessage('refreshLibrary', 'hyperplay')
-    } catch (error) {
-      logInfo(`Error while extracting game ${error}`, LogPrefix.HyperPlay)
-      window.webContents.send('gameStatusUpdate', {
-        appName,
-        runner: 'hyperplay',
-        folder: destinationPath,
-        status: 'done'
-      })
-      return { status: 'error', error: `${error}` }
+    if (!platformInfo.executable) {
+      return {
+        status: 'error',
+        error: 'Executable not found during install in HyperPlay game manager'
+      }
     }
-    return { status: 'done' }
+
+    return await extract(appName, {
```

Comment:
Not sure if we need `await` here since no other operation will run afterward

Reply:
True, although eslint will complain than, generally it should return extract(...) and since the function is async, than it will await for it still as it will return a promise
![image](https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/assets/75917402/fc9129ce-9d0e-4fce-9a81-6bdfb6efe46d)

Reply:
Our eslint config is going too far 😂

Reply:
Sure is 😆 I was thinking when I get that warning `why, but why` 🤣